msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfiguration.kt

```diff
@@ -0,0 +1,204 @@
+                )
+            }
+        }
+
+        if (defaultAuthority !is NativeAuthCIAMAuthority) {
+            // If the authority supplied is a CIAMAuthority, we can continue
+            if (defaultAuthority is CIAMAuthority) {
+                val nativeAuthAuthority = NativeAuthCIAMAuthority(
+                    authorityUrl = defaultAuthority.authorityUri.toString(),
+                    clientId = clientId
+                )
+                authorities.clear()
+                authorities.add(nativeAuthAuthority)
+            } else {
+                // If not, throw exception
+                throw MsalClientException(
+                    MsalClientException.NATIVE_AUTH_INVALID_CIAM_AUTHORITY_ERROR_CODE,
+                    MsalClientException.NATIVE_AUTH_INVALID_CIAM_AUTHORITY_ERROR_MESSAGE
+                )
+            }
+        }
+
+        // Enforce that shared device mode is disabled
+        if (isSharedDevice) {
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_SHARED_DEVICE_MODE_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_SHARED_DEVICE_MODE_ERROR_MESSAGE
+            )
+        }
+
+        // Enforce that dev cannot set "broker_redirect_uri_registered = true"
+        // This check is technically not necessary, since Native Auth configuration only loads with a
+        // CIAM authority, which does not result in creating a broker controller
+        // Adding it for clarity with devs
+        if (useBroker) {
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_ATTEMPTING_TO_USE_BROKER_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_ATTEMPTING_TO_USE_BROKER_ERROR_MESSAGE
+            )
+        }
+
+        // Check that challenge types are all valid
+        validateChallengeTypes()
+    }
+
+    /**
+     * Validates that the challenge types passed are valid
+     */
+    private fun validateChallengeTypes() {
+        // Make all challenge types lowercase for simplicity
+        challengeTypes = challengeTypes?.map { it.lowercase() }
+
+        challengeTypes?.forEach { challengeType ->
+            // Make sure challenge types passed were valid
+            if (challengeType !in VALID_CHALLENGE_TYPES) {
+                throw MsalClientException(
+                    MsalClientException.NATIVE_AUTH_INVALID_CHALLENGE_TYPE_ERROR_CODE,
+                    MsalClientException.NATIVE_AUTH_INVALID_CHALLENGE_TYPE_ERROR_MESSAGE + " \"" + challengeType + "\""
+                )
+            }
+        }
+    }
+
+    /**
+     * Overriding this method to add a check for redirect uri. If no uri was passed, we don't need to check this.
+     */
+    @Throws(MsalClientException::class)
+    override fun checkIntentFilterAddedToAppManifestForBrokerFlow() {
+        // TODO wonder if we really even need this, as broker is not used so we might also pass the check anyway.
```

Comment:
You still need this. The word "Broker" in that method name is a bit misleading. That method ensures that your app can handle redirection from Browser to your app for authorization code grant flow and you need to do that check if you want to support Browser fallback